### PR TITLE
Adds learn css functions

### PR DESCRIPTION
**Pull request by rachelandrew**
Fixes: #4846 Adding Learn CSS Functions.

### CODE REVIEW

File: src/site/content/en/learn/css/functions/index.md [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+```css
+.my-element {
+	width: calc(100% - 2rem);
+}
+```
+
+In this example, the `calc()` function is being used to size an element's width
+as 100% of its containing parent element,
+then removing `2rem` off that computed value.
+
```
Comment (una) [inline src/site/content/en/learn/css/functions/index.md:174]:
```suggestion
{% Codepen {
  user: 'web-dot-dev',
  id: 'PopYema'
} %}
```

made this calc demo

File: src/site/content/en/learn/css/functions/index.md [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+  id: 'MWJOWzP'
+} %}
+
+The [`rotate3d()`](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/rotate3d())
+function takes four arguments.
+The first 3 arguments are numbers, which define the X, Y and Z coordinates.
+The fourth argument is the rotation which,
+like the other rotation functions, accepts degrees, angle and turns.
+
+.my-element {
```
Comment (una) [inline src/site/content/en/learn/css/functions/index.md:306]:
```suggestion
```css
.my-element {
```

File: src/site/content/en/learn/css/functions/index.md [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+  transform: rotate3d(1, 1, 1, 10deg);
+}
+
```
Comment (una) [inline src/site/content/en/learn/css/functions/index.md:309]: Need to add ``` here to close the block, but the preview is confused by me adding it in suggested

File: src/site/content/en/learn/css/functions/index.md [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+along with `perspective-origin-x` and `perspective-origin-y` properties to create truly 3D experiences.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'bMqZmr'
+} %}
+
+## Animation functions, gradients and filters
+
+CSS also provides functions that help you animate elements,
```
Comment (una) [inline src/site/content/en/learn/css/functions/index.md:413]:
```suggestion
CSS also provides functions that help you [animate](/learn/css/animation) elements,
```

File: src/site/content/en/learn/css/functions/index.md [hunk 5]
```diff
@@ -0,0 +1,417 @@
...
+apply gradients to them and use graphical filters to manipulate how they look.
```
Comment (una) [inline src/site/content/en/learn/css/functions/index.md:414]:
```suggestion
apply [gradients](/learn/css/gradients) to them and use graphical [filters](/learn/css/filters) to manipulate how they look.
```

File: src/site/content/en/learn/css/functions/index.md [hunk 6]
```diff
@@ -0,0 +1,417 @@
...
+To keep this lesson as concise as possible,
+we're going to cover them in detail in the next few modules.
```
Comment (una) [inline src/site/content/en/learn/css/functions/index.md:416]:
```suggestion
they are covered in the linked modules.
```

**Review by una — APPROVED** [review]
Most of these are just add to batch and go, except the one where you'll need to PR and close the code block

File: src/site/content/en/learn/css/functions/index.md [hunk 7]
```diff
@@ -0,0 +1,417 @@
...
+in this lesson, you'll get  a high-level overview,
+giving you a much better understanding of where and how to use them.
+
+## What is a function?
+
+A function is a named, self-contained piece of code that completes a specific task.
+A function is named so you can call it within your code and you can pass data into the function.
+This is known as passing arguments.
+
+{% Img
```
[the remark this answers is not in the document]
Comment (rachelandrew) [reply src/site/content/en/learn/css/functions/index.md:37]: I think I'd probably prefer to change the image.

[the remark this answers is not in the document]
Comment (andy-piccalilli) [reply src/site/content/en/learn/css/functions/index.md:37]: Agreed. I've just pushed an update

**Pull request merged**